src/lib/comms/Makefile

```diff
@@ -33,7 +33,9 @@ obj-networking-$(PLATFORM_LINUX) += \
 
 obj-networking-$(COAP) += \
     coap.o \
-    sol-coap.o
+    sol-coap.o \
+    sol-coap-transport.o \
+    sol-coap-transport-ip.o
```

Comment:
with your change we should make `ip` (or actually udp) conditional to that, not `$(COAP)`, then it would be more like:

```
ifeq (y,$(COAP))
obj-networking-$(UDP) += sol-coap-transport-udp.o
endif
```

We'd need platforms to declare if they have UDP or not, currently they say "networking", but I guess we should be more explicit. What do you think, @ibriano?

Also, I'm assuming udp here and in your file, but if CoAP over TCP "just works", then you can keep that file with such name and call it general. However, my impression is that for TCP you'd need some different code, since there is no framing, and for coap purposes very likely we can implement different methods to say if the message needs a confirmation or not (I see this is not in the code yet, but with time we should optimize that case for protocols that guarantee delivery)